packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/PlanUpgradeDetails.tsx

```diff
@@ -65,9 +75,23 @@ export const PlanUpgradeDetails = ({
         {showTax && !!subTotal && !!exclusiveTaxRates.length && (
           <>
             <div className="plan-details-item">
-              <Localized id="plan-details-list-price">
-                <div>List Price</div>
-              </Localized>
+              {immediatelyInvoice ? (
+                <Localized id="plan-details-list-price">
+                  <div>List Price</div>
+                </Localized>
+              ) : (
+                <div>
+                  {productDetails.name || product_name} (
+                  {l10n.getString(
+                    `sub-update-interval-${formattedInterval}`,
+                    null,
+                    formattedInterval
+                      .toLowerCase()
+                      .replace(/\w/, (firstLetter) => firstLetter.toUpperCase())
+                  )}
+                  )
```

Comment:
Maybe I'm reading the code wrong, but above on L34 it looks like we're hardcoding `intervalCount: 1` to the `formatPlanInterval()`, which means we'd only get the four possible values `"daily"` | `"weekly"` | `"monthly"` | `"yearly"`.
Then we're trying to match the following localized ID:

```sh
`sub-update-interval-${formattedInterval}`
```

Why do we need to lowercase and then uppercase the first char if they're already localized w/ sentence casing in the .ftl files?

```ini
sub-update-interval-daily = Daily
sub-update-interval-weekly = Weekly
sub-update-interval-monthly = Monthly
sub-update-interval-yearly = Yearly
```

Ah&hellip; unless the string is the fallback value.

---

```sh
git grep "sub-update-interval-" | cat

packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/PlanUpgradeDetails.tsx:                    `sub-update-interval-${formattedInterval}`,
packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/en.ftl:sub-update-interval-daily = Daily
packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/en.ftl:sub-update-interval-weekly = Weekly
packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/en.ftl:sub-update-interval-monthly = Monthly
packages/fxa-payments-server/src/routes/Product/SubscriptionUpgrade/en.ftl:sub-update-interval-yearly = Yearly
```